src/mlpack/core/data/load_csv.hpp

```diff
@@ -367,7 +476,11 @@ class LoadCSV
             << ") on line " << col << "; should be " << rows << " dimensions.";
         throw std::runtime_error(oss.str());
       }
-
+      
+      // I am not able to understand when can we enter this case.
+      // I am looking into it, if anyone can give me some hint
+      // it might help, currently I've assigned canParser as true
+      // by default
```

Comment:
Try to test the master branch on a corrupted csv file. For instance, the first line should have commas and the second line is spaces. Try a mix of both of them on the same line too.
I do think this case should throw this error. Let me know if it is not the case:+1:

Reply:
Nope, both the cases are giving errors but they are from `canParse`.
Tried these
```
Input
7,6,3,6
3 4 8 6
2,8,4,6
8,2,3,6

Output
[WARN ] LoadCSV::TransposeParse(): wrong number of dimensions (1) on line 1; should be 4 dimensions.
0 1 6.95269e-310 6.95269e-310
6 6.95269e-310 2.11741e-317 6.92066e-310
3 4.94066e-324 3.21143e-322 0
6 6.95269e-310 2.12203e-314 4.94066e-324
```
```
Input
7,6,3,6
3,4 8 6
2,8,4,6
8,2,3,6

Output
[WARN ] LoadCSV::TransposeParse(): wrong number of dimensions (2) on line 1; should be 4 dimensions.
7 3 6.95257e-310 6.95257e-310
0 1 2.11741e-317 6.91434e-310
3 4.94066e-324 3.21143e-322 0
6 6.95257e-310 2.12203e-314 4.94066e-324
```

Reply:
@shrit should I proceed to remove the `canParse` part? You already told me that on but just wanted to confirm once again.

Reply:
@heisenbuug Yes, you can remove the `canParse` part. But we need to have some kind of a check that stop the parsing process if the dataset is corrupted, or if there is an issue with the col numbers. I did not look at the code yet, Let me know if such a check does exist

Reply:
If we `Load` with `DatasetMapper` then we are getting the warning as above in `new-parser` as well.
But when we are loading without `DatasetMapper` we are getting this as output.
```
7 3 2 8
6 4 8 2
3 0 4 3
6 0 6 6
```
This goes for both branches, `mlpack-master` and `new-parser`